src/me/mrCookieSlime/Slimefun/listeners/ArmorListener.java

```diff
@@ -32,48 +34,73 @@ public ArmorListener(SlimefunPlugin plugin) {
 		plugin.getServer().getPluginManager().registerEvents(this, plugin);
 	}
 	
-	@EventHandler(priority=EventPriority.MONITOR)
+	@EventHandler(priority = EventPriority.MONITOR)
 	public void onDamage(EntityDamageEvent e) {
 		if (e.getEntity() instanceof Player && !e.isCancelled()) {
 			Player p = (Player) e.getEntity();
 			for (ItemStack armor: p.getInventory().getArmorContents()) {
-				if (armor != null && SlimefunItem.getByItem(armor) != null) {
-					if (SlimefunItem.getByItem(armor).isItem(SlimefunItems.ENDER_BOOTS) && Slimefun.hasUnlocked(p, SlimefunItems.ENDER_BOOTS, true)) {
-						if (e instanceof EntityDamageByEntityEvent && ((EntityDamageByEntityEvent) e).getDamager() instanceof EnderPearl) e.setCancelled(true);
+				SlimefunItem item = SlimefunItem.getByItem(armor);
+				if (armor != null && item != null) {
+					if (item.isItem(SlimefunItems.ENDER_BOOTS)
+						&& Slimefun.hasUnlocked(p, SlimefunItems.ENDER_BOOTS, true)
+						&& e instanceof EntityDamageByEntityEvent
+						&& ((EntityDamageByEntityEvent) e).getDamager() instanceof EnderPearl
+					) {
+						e.setCancelled(true);
 					}
-					else if (SlimefunItem.getByItem(armor).isItem(SlimefunItems.SLIME_BOOTS) && Slimefun.hasUnlocked(p, SlimefunItems.SLIME_BOOTS, true)) {
-						if (e.getCause() == DamageCause.FALL) e.setCancelled(true);
+					else if (item.isItem(SlimefunItems.SLIME_BOOTS)
```

Comment:
Comparing the ItemStacks seems a bit harsh, you should just compare the id instead, much lighter.